### PR TITLE
Add supply model, controller, and routes

### PR DESCRIPTION
**Pull request by fisher-alice**
This PR adds the supply model, controller, and routes.
It also removes 'api' from the teacher endpoints.

### CODE REVIEW

**Review by seanypark505 — DISMISSED** [review]
Looks great, let's create a ticket so we remember to include controller function that returns all supplies from the teacher's supply array.

**Comment by fisher-alice** [comment]
> Looks great, let's create a ticket so we remember to include controller function that returns all supplies from the teacher's supply array.

Just pushed a commit that does this.

File: src/routes/teacher.route.js [hunk 1]
```diff
@@ -4,10 +4,11 @@ import { authJwt } from '../services/auth.service.js';
 
 const router = express.Router();
 
-router.route('/api/teachers').get(authJwt, teacherCtrl.list);
+router.route('/teachers').get(authJwt, teacherCtrl.list);
 
 router
-    .route('/api/teachers/:teacherId')
+    .route('/teachers/:teacherId')
+    .get(teacherCtrl.readPublic)
```
Comment (seanypark505) [inline src/routes/teacher.route.js]: Can't have two GET handlers under the same route path.  Creating a route specifically for public would work, something like:

```javascript
router.route('/teachers/:teacherId/public').get(teacherCtrl.readPublic)
```

**Review by seanypark505 — CHANGES_REQUESTED** [review]
See comment in teacher.route.js file

**Review by joelswenddal — APPROVED** [review]
End points seem right to me. Good work!

**Review by seanypark505 — APPROVED** [review]

**Pull request merged**